sfa_dash/blueprints/util.py

```diff
@@ -150,9 +153,20 @@ def get_forecast_table(cls, site_id=None, **kwargs):
             If a site_id is passed and the user does not have access
             to that site or some other api error has occurred.
         """
-        creation_link = cls.creation_link('forecast', site_id)
+        request_kwargs = {}
+        if site_id is not None:
+            creation_link = cls.creation_link('forecast',
+                                              site_id=site_id)
+            request_kwargs['site_id'] = site_id
+        elif aggregate_id is not None:
+            creation_link = cls.creation_link('forecast',
+                                              aggregate_id=aggregate_id)
+            request_kwargs['aggregate_id'] = aggregate_id
+        else:
+            creation_link = None
+
         forecast_data = handle_response(
-            forecasts.list_metadata(site_id=site_id))
+            forecasts.list_metadata(**request_kwargs))
```

Comment:
passing the dict like this obfuscates what's going on. You really don't need request_kwargs at all I think and can just do ``forecasts.list_metadata(site_id=site_id, aggregate_id=aggregate_id)